src/sctools/metrics/aggregator.py

```diff
@@ -333,6 +331,86 @@ def parse_molecule(
             # todo figure out antisense and make this notation clearer; info likely in dropseqtools
             self._plus_strand_reads += not record.is_reverse
 
+    def parse_molecule_fast(
+        self, tags: Sequence[str], records: List[str]
+    ) -> None:
+        """Parse information from all records of a molecule.
+
+        The parsed information is stored in the MetricAggregator in-place.
+
+        Parameters
+        ----------
+        tags : Sequence[str]
+            all the tags that define this molecule. one of {[CB, GE, UB], [GE, CB, UB]}
+        records : List[(str, str, str, str, str)]
+            the records associated with the molecule
+
+        """
+        for record in records:
+            # todo think about how I could use the duplicate tag to reduce computation; duplicates
+            # should normally come in order in a sorted file
+
+            # extract sub-class-specific information
+            self.parse_extra_fields_fast(tags=tags, record=record)
+
+            #if tags[0]=='TAAACCGCAAACTGTC':
+            #    print(tags)
+            self.n_reads += 1
+            # self.noise_reads += self.is_noise(record)  # todo implement me
```

Comment:
Remove this.

Reply:
removed

Reply:
same here, all python code is set back to the repo.